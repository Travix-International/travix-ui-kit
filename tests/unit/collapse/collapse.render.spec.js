import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import Collapse from '../../../components/collapse/collapse';
import CollapseItem from '../../../components/collapse/collapseItem';

describe('Collapse: render', () => {
  it('should return empty collapse component', () => {
    const component = shallow(
      <Collapse />
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should return collapse component with two items', () => {
    const component = shallow(
      <Collapse>
        <CollapseItem title="Collapse Item 1">
          <p>Collapse content 1</p>
        </CollapseItem>
        <CollapseItem title="Collapse Item 2">
          <p>Collapse content 2</p>
        </CollapseItem>
      </Collapse>
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should return "accordion" collapse component with two items', () => {
    const component = shallow(
      <Collapse isAccordion>
        <CollapseItem title="Collapse Item 1">
          <p>Collapse content 1</p>
        </CollapseItem>
        <CollapseItem title="Collapse Item 2">
          <p>Collapse content 2</p>
        </CollapseItem>
      </Collapse>
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should not render collapse component child if it not "CollapseItem"', () => {
    const component = shallow(
      <Collapse isAccordion>
        <div>
          This item should not be rendered
        </div>
        <CollapseItem title="Collapse Item 2">
          <p>Collapse content 2</p>
        </CollapseItem>
      </Collapse>
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render collapse component items with custom id', () => {
    const component = shallow(
      <Collapse isAccordion>
        <CollapseItem id="сid1" title="Collapse Item 1">
          <p>Collapse content 1</p>
        </CollapseItem>
        <CollapseItem id="сid2" title="Collapse Item 2">
          <p>Collapse content 2</p>
        </CollapseItem>
      </Collapse>
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render collapse component items and first item should be expanded', () => {
    const component = shallow(
      <Collapse activeKey="сid1" isAccordion>
        <CollapseItem id="сid1" title="Collapse Item 1">
          <p>Collapse content 1</p>
        </CollapseItem>
        <CollapseItem id="сid2" title="Collapse Item 2">
          <p>Collapse content 2</p>
        </CollapseItem>
      </Collapse>
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render collapse component with expanded item by default', () => {
    const component = shallow(
      <Collapse defaultActiveKey="сid2" isAccordion>
        <CollapseItem id="сid1" title="Collapse Item 1">
          <p>Collapse content 1</p>
        </CollapseItem>
        <CollapseItem id="сid2" title="Collapse Item 2">
          <p>Collapse content 2</p>
        </CollapseItem>
      </Collapse>
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render collapse component with expanded item based on props when "defaultActiveKey" is provided', () => {
    const component = shallow(
      <Collapse activeKey="сid1" defaultActiveKey="сid2" isAccordion>
        <CollapseItem id="сid1" title="Collapse Item 1">
          <p>Collapse content 1</p>
        </CollapseItem>
        <CollapseItem id="сid2" title="Collapse Item 2">
          <p>Collapse content 2</p>
        </CollapseItem>
      </Collapse>
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render collapse component with expanded items based on "defaultActiveKey"', () => {
    const component = shallow(
      <Collapse defaultActiveKey={['сid1', 'сid2']}>
        <CollapseItem id="сid1" title="Collapse Item 1">
          <p>Collapse content 1</p>
        </CollapseItem>
        <CollapseItem id="сid2" title="Collapse Item 2">
          <p>Collapse content 2</p>
        </CollapseItem>
      </Collapse>
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
