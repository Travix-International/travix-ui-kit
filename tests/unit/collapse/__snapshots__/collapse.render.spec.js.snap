// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse: render should not render collapse component child if it not "CollapseItem" 1`] = `
<div
  className="ui-collapse ui-collapse_accordion"
>
  <CollapseItem
    id="ui-collapse-item.1"
    isActive={false}
    onClick={[Function]}
    title="Collapse Item 2"
  >
    <p>
      Collapse content 2
    </p>
  </CollapseItem>
</div>
`;

exports[`Collapse: render should render collapse component items and first item should be expanded 1`] = `
<div
  activeKey="сid1"
  className="ui-collapse ui-collapse_accordion"
>
  <CollapseItem
    id="сid1"
    isActive={true}
    onClick={[Function]}
    title="Collapse Item 1"
  >
    <p>
      Collapse content 1
    </p>
  </CollapseItem>
  <CollapseItem
    id="сid2"
    isActive={false}
    onClick={[Function]}
    title="Collapse Item 2"
  >
    <p>
      Collapse content 2
    </p>
  </CollapseItem>
</div>
`;

exports[`Collapse: render should render collapse component items with custom id 1`] = `
<div
  className="ui-collapse ui-collapse_accordion"
>
  <CollapseItem
    id="сid1"
    isActive={false}
    onClick={[Function]}
    title="Collapse Item 1"
  >
    <p>
      Collapse content 1
    </p>
  </CollapseItem>
  <CollapseItem
    id="сid2"
    isActive={false}
    onClick={[Function]}
    title="Collapse Item 2"
  >
    <p>
      Collapse content 2
    </p>
  </CollapseItem>
</div>
`;

exports[`Collapse: render should render collapse component with expanded item based on props when "defaultActiveKey" is provided 1`] = `
<div
  activeKey="сid1"
  className="ui-collapse ui-collapse_accordion"
  defaultActiveKey="сid2"
>
  <CollapseItem
    id="сid1"
    isActive={true}
    onClick={[Function]}
    title="Collapse Item 1"
  >
    <p>
      Collapse content 1
    </p>
  </CollapseItem>
  <CollapseItem
    id="сid2"
    isActive={false}
    onClick={[Function]}
    title="Collapse Item 2"
  >
    <p>
      Collapse content 2
    </p>
  </CollapseItem>
</div>
`;

exports[`Collapse: render should render collapse component with expanded item by default 1`] = `
<div
  className="ui-collapse ui-collapse_accordion"
  defaultActiveKey="сid2"
>
  <CollapseItem
    id="сid1"
    isActive={false}
    onClick={[Function]}
    title="Collapse Item 1"
  >
    <p>
      Collapse content 1
    </p>
  </CollapseItem>
  <CollapseItem
    id="сid2"
    isActive={true}
    onClick={[Function]}
    title="Collapse Item 2"
  >
    <p>
      Collapse content 2
    </p>
  </CollapseItem>
</div>
`;

exports[`Collapse: render should render collapse component with expanded items based on "defaultActiveKey" 1`] = `
<div
  className="ui-collapse"
  defaultActiveKey={
    Array [
      "сid1",
      "сid2",
    ]
  }
>
  <CollapseItem
    id="сid1"
    isActive={true}
    onClick={[Function]}
    title="Collapse Item 1"
  >
    <p>
      Collapse content 1
    </p>
  </CollapseItem>
  <CollapseItem
    id="сid2"
    isActive={true}
    onClick={[Function]}
    title="Collapse Item 2"
  >
    <p>
      Collapse content 2
    </p>
  </CollapseItem>
</div>
`;

exports[`Collapse: render should return "accordion" collapse component with two items 1`] = `
<div
  className="ui-collapse ui-collapse_accordion"
>
  <CollapseItem
    id="ui-collapse-item.0"
    isActive={false}
    onClick={[Function]}
    title="Collapse Item 1"
  >
    <p>
      Collapse content 1
    </p>
  </CollapseItem>
  <CollapseItem
    id="ui-collapse-item.1"
    isActive={false}
    onClick={[Function]}
    title="Collapse Item 2"
  >
    <p>
      Collapse content 2
    </p>
  </CollapseItem>
</div>
`;

exports[`Collapse: render should return collapse component with two items 1`] = `
<div
  className="ui-collapse"
>
  <CollapseItem
    id="ui-collapse-item.0"
    isActive={false}
    onClick={[Function]}
    title="Collapse Item 1"
  >
    <p>
      Collapse content 1
    </p>
  </CollapseItem>
  <CollapseItem
    id="ui-collapse-item.1"
    isActive={false}
    onClick={[Function]}
    title="Collapse Item 2"
  >
    <p>
      Collapse content 2
    </p>
  </CollapseItem>
</div>
`;

exports[`Collapse: render should return empty collapse component 1`] = `null`;
